vpr/src/route/heap_type.h

```diff
@@ -43,11 +43,36 @@ struct t_heap {
 
     struct t_prev {
         int node;
-        int edge;
+        unsigned int edge;
```

Comment:
Ultimately this should be replaceable with just RREdgeId (as noted earlier). Thoughts?

Reply:
Having non-trival types in a union is hard(-ish).  It is possible, but it is really intended for things like a variant type.  I used this approach because it avoided needing to handle non-trival types under the union